oracle/src/utils/constants.js

```diff
@@ -27,5 +27,17 @@ module.exports = {
   FALLBACK_RPC_URL_SWITCH_TIMEOUT: 60 * 60 * 1000,
   SENDER_QUEUE_MAX_PRIORITY: 10,
   SENDER_QUEUE_SEND_PRIORITY: 5,
-  SENDER_QUEUE_CHECK_STATUS_PRIORITY: 1
+  SENDER_QUEUE_CHECK_STATUS_PRIORITY: 1,
+  ASYNC_CALL_ERRORS: {
+    // requested transaction/block/receipt does not exist
+    NOT_FOUND: '0x0000000000000000000000000000000000000000000000000000000000000000',
+    // requested custom block does not exist yet or its timestamp is greater than the home block timestamp
+    BLOCK_IS_IN_THE_FUTURE: '0x0000000000000000000000000000000000000000000000000000000000000001',
+    // eth_call has reverted or finished with OOG error
+    REVERT: '0x0000000000000000000000000000000000000000000000000000000000000002',
+    // evaluated output length exceeds allowed length of 64 KB
+    RESULT_IS_TOO_LONG: '0x0000000000000000000000000000000000000000000000000000000000000003'
+  },
+  MAX_ASYNC_CALL_RESULT_LENGTH: 64 * 1024,
+  ASYNC_ETH_CALL_MAX_GAS_LIMIT: 100000000
```

Comment:
Do you think to have a lower gas limit will allow to avoid ddos attacks on the RPC nodes? Since several oracles will requests the same request at the same time it could insult the nodes.

Reply:
I know that Infura uses 10x latest block gas limit as a cap for eth_call/eth_estimateGas methods in order to prevent such kind of abuse. So, I think we are quite safe to go with 100.000.000 gas limit.